packages/js/create-product-editor-block/plugin-templates/$slug.php.mustache

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Plugin Name:          {{title}}
+{{#pluginURI}}
+ * Plugin URI:           {{{pluginURI}}}
+{{/pluginURI}}
+{{#description}}
+ * Description:          {{description}}
+{{/description}}
+ * Version:              {{version}}
+ * Requires at least:    6.2
+ * WC requires at least: 7.8
+ * Requires PHP:         7.3
+{{#author}}
+ * Author:               {{author}}
+{{/author}}
+{{#license}}
+ * License:              {{license}}
+{{/license}}
+{{#licenseURI}}
+ * License URI:          {{{licenseURI}}}
+{{/licenseURI}}
+ * Text Domain:          {{textdomain}}
+{{#domainPath}}
+ * Domain Path:          {{{domainPath}}}
+{{/domainPath}}
+{{#updateURI}}
+ * Update URI:           {{{updateURI}}}
+{{/updateURI}}
+ *
+ * @package              {{namespace}}
+ */
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function {{namespaceSnakeCase}}_{{slugSnakeCase}}_block_init() {
+	if ( isset( $_GET['page'] ) && $_GET['page'] === 'wc-admin' ) {
+		register_block_type( __DIR__ . '/build' );
+	}
+}
+add_action( 'init', '{{namespaceSnakeCase}}_{{slugSnakeCase}}_block_init' );
+
+function {{namespaceSnakeCase}}_{{slugSnakeCase}}_add_block_to_product_editor( $args ) {
+	$template = $args['template'];
```

Comment:
Now that our template addition is only added when the feature flag is enabled, we should just add a check here to see if `template` is defined.
At some point we may want to check for the for the feature flag as well.
So could add this:
```
if ( ! isset( $args['template'] ) ) {
  return $args;
}
```
We could add these conditions:
```
class_exists( '\Automattic\WooCommerce\Utilities\FeaturesUtil' ) && \Automattic\WooCommerce\Utilities\FeaturesUtil::feature_is_enabled('product_block_editor')
```